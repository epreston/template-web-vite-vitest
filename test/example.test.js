import { beforeAll, beforeEach, describe, expect, it, vi } from 'vitest';

// ---------------------------------------------------------

// import { itemsToMock } from '../src/testDependency';

// ---------------------------------------------------------

// vi.mock('../src/testDependency');

// ---------------------------------------------------------

// import { itemBeingTested } from '../src/testTarget';

// ---------------------------------------------------------

const props = {
  foo: vi.fn(),
  bar: {
    baz: true,
  },
};

// ---------------------------------------------------------

describe('Object', () => {
  let instance;

  beforeAll(() => {
    instance = Object.assign({}, props);
  });

  // beforeEach(() => {
  //   instance = Object.assign({}, props);
  // });

  it('should expose an object', () => {
    expect(instance).toBeDefined();
    expect(instance).toBeInstanceOf(Object);
  });

  it('should have the expected properties', () => {
    expect(instance).toHaveProperty('bar');
  });

  it('should have a method foo()', () => {
    // instance.foo();
    expect(instance.foo).toBeDefined();

    instance.foo();
    expect(instance.foo).toHaveBeenCalled();
  });

  it('should load and configure the object correctly', () => {
    expect(instance).toMatchSnapshot();
  });
});
