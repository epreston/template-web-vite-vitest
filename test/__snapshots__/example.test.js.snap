// Vitest Snapshot v1

exports[`Object > should load and configure the object correctly 1`] = `
{
  "bar": {
    "baz": true,
  },
  "foo": [MockFunction spy] {
    "calls": [
      [],
    ],
    "results": [
      {
        "type": "return",
        "value": undefined,
      },
    ],
  },
}
`;
